File: src/003.NumberAndPlace/009.PalindromeNumber/9.回文数.go
/*
 * @lc app=leetcode.cn id=9 lang=golang
 *
 * [9] 回文数
 */

// @lc code=start
// 数学思想
func isPalindrome(x int) bool {
	if x < 0 || (x % 10 == 0 && x != 0) {
		return false
	}

	cur := 0
	tempX := x
	for tempX != 0 {
		cur = cur * 10 + tempX % 10
		tempX = tempX / 10
	}

	return x == cur
}
// @lc code=end

File: src/003.NumberAndPlace/009.PalindromeNumber/9.回文数.js
/*
 * @lc app=leetcode.cn id=9 lang=javascript
 *
 * [9] 回文数
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */

// 双指针思想 
var isPalindrome = function(x) {
  if (x < 0) false;
  let strX = '' + x;
  let l = 0, r = strX.length - 1;

  while (l <= r) {
    if (strX[l] !== strX[r]) return false;
    l++;
    r--;
  }

  return true;
};
// @lc code=end

